packages/expo-cli/src/commands/utils/npm.ts

```diff
@@ -69,42 +73,37 @@ export async function npmViewAsync(...props: string[]): Promise<JSONValue> {
   }
 }
 
-/** Given a package name like `expo` or `expo@beta`, return the registry URL if it exists. */
-export async function getNpmUrlAsync(packageName: string): Promise<string> {
-  const results = await npmViewAsync(packageName, 'dist.tarball');
-
-  assert(results, `Could not get npm url for package "${packageName}"`);
-
-  // Fully qualified url returns a string.
-  // Example:
-  // 𝝠 npm view expo-template-bare-minimum@sdk-33 dist.tarball --json
-  if (typeof results === 'string') {
-    return results;
-  }
-
-  // When the tag is arbitrary, the tarball url is an array, return the last value as it's the most recent.
-  // Example:
-  // 𝝠 npm view expo-template-bare-minimum@33 dist.tarball --json
-  if (Array.isArray(results)) {
-    return results[results.length - 1] as string;
-  }
-
-  throw new CommandError(
-    'Expected results of `npm view ...` to be an array or string. Instead found: ' + results
-  );
-}
-
 // @ts-ignore
 const pipeline = promisify(Stream.pipeline);
 
 export async function downloadAndExtractNpmModuleAsync(
   npmName: string,
   props: ExtractProps
 ): Promise<void> {
-  const url = await getNpmUrlAsync(npmName);
-
-  Log.debug('Fetch from URL:', url);
-  await extractNpmTarballFromUrlAsync(url, props);
+  Log.debug(`Looking for tarball for ${npmName} in ${getCacheFilePath()}...`);
+  try {
+    const cachePath = getCacheFilePath();
+    // Perform dry-run to get actual filename for resolved version
+    const filename = (((await npmPackAsync(
+      npmName,
+      cachePath,
+      '--dry-run'
+    )) as JSONArray)[0] as JSONObject).filename as string;
+
+    const cacheFilename = path.join(cachePath, filename);
+
+    // TODO: This cache does not expire, but neither does the FileCache at the top of this file.
+    if (!(await fs.stat(cacheFilename).catch(() => null))?.isFile() ?? false) {
+      Log.debug(`Downloading tarball for ${npmName} to ${cachePath}...`);
+      await npmPackAsync(npmName, cachePath);
+    }
+    await extractLocalNpmTarballAsync(cacheFilename, {
+      cwd: props.cwd,
+      name: npmName,
+    });
+  } catch (error) {
+    Log.error('Error downloading and extracting template package');
```

Comment:
```suggestion
    Log.error('Error downloading and extracting template package:', error);
```
This could provide more context.